syntax = "proto3";

package helloworld.v1;

service Greeter {
  // 單次請求-回應
  rpc SayHello (HelloRequest) returns (HelloReply);

  // 雙向串流：兩邊都可連續傳訊息
  rpc Chat (stream ChatMessage) returns (stream ChatMessage);
}

message HelloRequest { string name = 1; }
message HelloReply   { string message = 1; }

message ChatMessage {
  string sender = 1;
  string text   = 2;
  int64  ts     = 3;
}